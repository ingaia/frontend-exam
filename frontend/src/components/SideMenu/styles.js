import styled from 'styled-components';
import { lighten } from 'polished';

export const Menu = styled.div`
    display: flex;
    flex-direction: column;
    min-width: 423px;
    border-right: 1px solid #4c4c4c;
    justify-content: flex-start;
    align-items: center;
    height: 100%;
    img {
        margin-top: 100px;
    }

    button {
        width: 254px;
        height: 39px;
        background: transparent;
        margin-top: 130px;
        border: 1px solid #a99e7e;
        font-size: 14px;
        line-height: 16px;
        color: #a99e7e;
        transition: background 0.2s;

        &:hover {
            background: ${lighten(0.9, 'rgba(255, 255, 255, 0.1)')};
        }
    }

    span {
        margin-top: 30px;
        font-size: 14px;
        line-height: 16px;
        color: #fff;
        &:hover {
            cursor: pointer;
        }
    }
    @media (max-width: 992px) {
        display: none;
    }
`;
