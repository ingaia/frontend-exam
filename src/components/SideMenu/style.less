@import './../../variables.less';

.side-menu {
    display: none;
    flex: unset;
    position: sticky;
    top: 0;
    padding: 10px 5px;
    background: rgba(0, 0, 0, 0.9);
    border-bottom: 2px solid @hillary;
    z-index: 10;
    
    .side-menu-header {
        display: flex;
        align-items: center;
        justify-content: center;
        position: relative;
    }

    .fa-bars, .fa-times {
        position: absolute;
        left: 0;
        padding-left: 10px;
        color:#fff;
        font-size: 30px;
    }

    .logo-box {
        width: 100%;
        text-align: center;
    }

    .logo {
        width: 130px;
    }

    .sidenav {
        display: flex;
        align-items: center;
        flex-direction: column;
        position: fixed;
        left: 0;
        top: 92px;
        height: 100%;
        width: 0;
        padding-top: 60px;
        background: rgba(0, 0, 0, 0.9);
        z-index: 1;
        overflow-x: hidden;
        transition: 0.5s;

        &.open {
            width: 250px;
        }
    }

    .btn {
        padding: 10px 50px;
    }
}


@media @mobile {
    .side-menu {
        display: inline;
    }
}