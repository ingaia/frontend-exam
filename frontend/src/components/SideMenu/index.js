import React from 'react';

import { Menu } from './styles';

import logo from '~/assets/logo_dark.png';

export default function SideMenu() {
    return (
        <Menu>
            <img src={logo} alt="logo" width={274} height={137} />
            <button type="button">TRAILERS</button>
            <span>LOGOUT</span>
        </Menu>
    );
}
