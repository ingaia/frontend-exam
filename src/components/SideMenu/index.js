import React, { useState } from 'react';
import Button from '../../components/Button';

import logo_dark from '../../../assets/logo_dark.png';
import './style.less';

export default function SideMenu(props){
    const [open, setOpen] = useState(false);

    return (
        <div className="side-menu">
            <div className="side-menu-header">
                { open ? <i className="fas fa-times" onClick={() => setOpen(false)}></i> 
                    :   <i className="fas fa-bars" onClick={() => setOpen(true)}></i>
                }
                <div className="logo-box">
                    <img className="logo" src={logo_dark} alt=""/>
                </div>
            </div>
            <div className={`sidenav ${open && 'open'}`}>
                <Button className="btn-hillary" label="Trailers"/>
                <Button className="btn-default" label="Logout" onClick={props.logOut}/> 
            </div>
        </div>
    );
}