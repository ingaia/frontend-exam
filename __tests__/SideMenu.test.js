import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json'
import SideMenu from '../src/components/SideMenu';

describe('<SideMenu/>', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<SideMenu />);
    });

    it('should render', () => {
        expect(wrapper.hasClass('side-menu')).toBe(true);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should call the function logOut', () => {
        const logOut = jest.fn();
        wrapper.setProps({ logOut })
        wrapper.find('.btn-default').simulate('click');
        expect(logOut).toHaveBeenCalled();
    });
});