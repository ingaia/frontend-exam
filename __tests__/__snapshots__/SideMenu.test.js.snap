// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideMenu/> should render 1`] = `
<div
  className="side-menu"
>
  <div
    className="side-menu-header"
  >
    <i
      className="fas fa-bars"
      onClick={[Function]}
    />
    <div
      className="logo-box"
    >
      <img
        alt=""
        className="logo"
        src="test-file-stub"
      />
    </div>
  </div>
  <div
    className="sidenav false"
  >
    <Button
      className="btn-hillary"
      label="Trailers"
    />
    <Button
      className="btn-default"
      label="Logout"
    />
  </div>
</div>
`;
